spring-kafka/src/main/java/org/springframework/kafka/listener/ListenerMetadata.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.listener;
+
+/**
+ * Metadata associated to a {@link org.springframework.kafka.annotation.KafkaListener}.
+ *
+ * @author Francois Rosiere
+ * @since 2.8.5
+ * @see org.springframework.kafka.annotation.KafkaListener
+ */
+public interface ListenerMetadata {
+
+	/**
+	 * Return the listener id.
+	 * @return the listener id.
+	 */
+	String getListenerId();
+
+	/**
+	 * Return the group id.
+	 * @return the group id.
+	 */
+	String getGroupId();
```

Comment:
I think we should add `byte[] getListenerInfo()` (new property on `@KafkaListener` since 2.8.4).

Reply:
Good idea, field added.